bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/HSBType.java

```diff
@@ -269,7 +267,12 @@ public State as(Class<? extends State> target) {
         } else if (target == PercentType.class) {
             return new PercentType(getBrightness().toBigDecimal());
         } else {
-            return StateConverterUtil.defaultConversion(this, target);
+            // if the target is the same type return, otherwise give up
```

Comment:
could you please define a `protected` method in `NumberType` which there delegates to `State.super.as(target)`? Then you don't need to repeat his logic twice.